src/main/java/com/google/sps/servlets/CustomSearchResults.java

```diff
@@ -0,0 +1,242 @@
+package com.google.sps.servlets;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+import java.util.ArrayList;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+import com.joestelmach.natty.Parser;
+import com.joestelmach.natty.DateGroup;
+
+class CustomSearchResults {
```

Comment:
If I understood the code correctly, this class does two things:
1. parses the data
2. it's a container of the results too.

Maybe you should break it into two classes.